File: Python/python_fastapi/app/produto/produtoController.py
from fastapi import APIRouter
from app.schemas import CreateProdutoDto
from app.produto.produtoService import (
    criar_produto,
    listar_todos_produtos,
    buscar_por_categoria,
)

router = APIRouter(prefix="/produtos")

@router.post("/")
def create(dto: CreateProdutoDto):
    return criar_produto(dto)

@router.get("/")
def find_all():
    return listar_todos_produtos()

@router.get("/{categoria}")
def find_categoria(categoria: str):
    return buscar_por_categoria(categoria)

File: JavaScript/java_script/src/produto/produto.controller.ts
import { Controller, Post, Get, Body, Param } from '@nestjs/common';
import { ProdutosService } from './produto.service';
import { CreateProdutoDto } from '../create_.dto';

@Controller('produtos')
export class ProdutoController {
  constructor(private readonly produtosService: ProdutosService) {}

  @Post()
  async create(@Body() dto: CreateProdutoDto) {
    return this.produtosService.create(dto);
  }

  @Get()
  async findAll() {
    return this.produtosService.findAll();
  }

  @Get(':categoria')
  async findCategoria(@Param('categoria') categoria: string) {
    return this.produtosService.findCategoria(categoria);
  }
}
